exe/task.hpp

```diff
@@ -0,0 +1,44 @@
+#include <vector>
+#include <map>
+#include <set>
+#include <string>
+#include <functional>
+#include <stdexcept>
+
+struct task_error : std::runtime_error
+{
+    task_error(std::string t, std::string msg) :
+        std::runtime_error(std::move(msg)), type(std::move(t)) {}
+    std::string type;
+};
+
+struct task
+{
+    std::string name;
+    std::vector<std::string> args;
+};
+
+struct task_runner
+{
+    task find_task(std::string taskname);
+
+private:
+    std::set<std::string> reserved_extensions_{".json", ".md"};
+    // Hard-code extensions as executable on Windows. On non-Windows, we still rely on permissions and #!
+    std::map<std::string, std::function<task(std::string)>> builtin_tasks_{
```

Comment:
I wish there was a name for this field which would better correspond to its actual purpose, but I could not come up with anything which would be closer to the purpose yet short ...